xsl/xccdf-resources/openscap.js

```diff
@@ -108,12 +109,152 @@ function ruleSearch()
         $("#search-matches").html("No rules match your search criteria!");
 }
 
+var is_original = true;
+var original_treetable = null;
+
 $(document).ready( function() {
     $("#result-details").hide();
     $(".js-only").show();
+	$(".form-group select").val("default");
     $(".toggle-rule-display").each(function(){
         toggleRuleDisplay(this);
     });
-
+    original_treetable = $(".treetable").clone();
     $(".treetable").treetable({ column: 0, expandable: true, initialState : "expanded", indent : 0 });
+    is_original = true;
+	stripeTreeTable();
 });
+
+function resetTreetable() {
+	if (!is_original) {
+		$(".treetable").remove();
+		$("#rule-overview").append(original_treetable.clone());
+		$(".treetable").treetable({ column: 0, expandable: true, clickableNodeNames: true, initialState : "expanded", indent : 0 });
+		$(".toggle-rule-display").each(function(){
+			toggleRuleDisplay(this);
+		});
+		is_original = true;
+	}
+}
+
+function newGroupLine(group_name)
+{
+	return "<tr class=\"rule-overview-inner-node\" data-tt-id=\"" + group_name + "\">" +
+		"<td colspan=\"3\"><strong>" + group_name + "</strong></td></tr>";
+}
+
+var KeysEnum = {
+	DEFAULT: "default",
+	SEVERITY : "severity",
+	RESULT: "result",
+	NIST: "data-nist-id",
+	DISA: "data-disa-id"
+};
+
+/* This function returns an array of target groups indentifiers */
+function getTargetGroupsList(rule, key)
+{
+	switch(key) {
+	case KeysEnum.SEVERITY:
+		var severity = rule.children("td:nth-child(2)").text();
+		return [severity];
+	case KeysEnum.RESULT:
+		var result = rule.children(".rule-result").text();
+		return [result];
+	default:
+		var ref_list = rule.attr(key);
+		if (!ref_list) {
+			ref_list = "unknown";
+		}
+		return ref_list.split(",");
```

Comment:
What if the references contain `,` in them? Can that happen according to the XCCDF spec?